node/info.go

```diff
@@ -39,17 +49,24 @@ func (rn *ChordRemoteNodeInfo) MarshalJSON() ([]byte, error) {
 	}
 
 	nodeData := &pb.NodeData{}
-	err = proto.Unmarshal(rn.Node.Node.Data, nodeData)
+	err = proto.Unmarshal(crn.remoteNode.Node.Node.Data, nodeData)
 	if err != nil {
 		return nil, err
 	}
 
 	delete(out, "data")
-	out["id"] = hex.EncodeToString(rn.Node.Node.Id)
-	out["isOutbound"] = rn.IsOutbound
-	out["roundTripTime"] = (*nnetnode.RemoteNode)(rn).GetRoundTripTime() / time.Millisecond
+	out["id"] = hex.EncodeToString(crn.remoteNode.Node.Node.Id)
+	out["isOutbound"] = crn.remoteNode.IsOutbound
+	out["roundTripTime"] = crn.remoteNode.GetRoundTripTime() / time.Millisecond
 	out["protocolVersion"] = nodeData.ProtocolVersion
 
+	nbr := crn.localNode.getNeighborByNNetNode(crn.remoteNode)
+	if nbr != nil {
+		out["connTime"] = time.Since(nbr.Node.startTime) / time.Second
+	} else {
+		out["connTime"] = 0
```

Comment:
Does it make sense when `nbr == nil` and we still have node info like `roundTripTime`?

Reply:
In this case I prefer to mv L67 to L62

Reply:
Done